src/du_parameters/frequency.py

```diff
@@ -0,0 +1,644 @@
+            other (Any): The value to compare.
+
+        Returns:
+            bool: True if the current GSCN instance is greater than or equal to the other,
+             False otherwise.
+        """
+        if isinstance(other, GSCN):
+            return self._channel >= other._channel
+        if isinstance(other, int | Decimal):
+            return self._channel >= other
+        return False
+
+    def __sub__(self, other: Any) -> "GSCN":
+        """Subtract another GSCN or integer or decimal from this GSCN.
+
+        Args:
+            other (Any): The value to subtract.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(self._channel - other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(self._channel - round(other))
+        raise NotImplementedError(f"Unsupported type for subtraction: {type(other).__name__}")
+
+    def __add__(self, other: Any) -> "GSCN":
+        """Add another GSCN or integer or Decimal to this GSCN.
+
+        Args:
+            other (Any): The value to add.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(self._channel + other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(self._channel + round(other))
+        raise NotImplementedError(f"Unsupported type for addition: {type(other).__name__}")
+
+    def __truediv__(self, other: Any) -> "GSCN":
+        """Divide GSCN to other GSCN, integer or Decimal.
+
+        Args:
+            other (Any): The value to divide by.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(round(self._channel / other._channel))  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(round(self._channel / other))  # type: ignore[operator]
+        raise NotImplementedError(f"Unsupported type for division: {type(other).__name__}")
+
+    def to_frequency(self) -> Frequency:
+        """Calculate the frequency using input GSCN.
+
+        Args:
+           gscn (GSCN): The input GSCN.
+
+        Returns:
+            frequency (Frequency): The closest frequency.
+
+        Raises:
+            ValueError: If the GSCN is out of supported range or n is out of range.
+            TypeError: If the input is not a GSCN.
+        """
+        try:
+            config = get_range_from_gscn(self)
```

Comment:
```suggestion
            config = get_range_from_gscn(self._channel)
```
Applies to all occurrences of `self` in this method.